### PR TITLE
feat: Enhance ai-cache Plugin with Vector Similarity-Based LLM Cache Recall and Multi-DB Support

### PR DESCRIPTION
**Pull request by EnableAsync**
### Ⅰ. Describe what this PR did
This PR extends the functionality of the `ai-cache` plugin, enabling more efficient AI application development by introducing vector similarity-based caching and recall mechanisms.

### Ⅱ. Does this pull request fix one issue?

Please refer to issue #1040 and #1041.

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
Test cases will be added later.

### Ⅳ. Describe how to verify it
After filling in the `apikey` and `ChromaCollectionID` in `docker-compose-test/envoy.yaml`, execute the following code:
```bash
cd docker-compose-test/
docker compose up
```
Then test it by accessing the LLM via cURL:
```bash
curl http://172.17.0.1:10000/v1/chat/completions -X POST -d '{"model":"gpt-4o-mini","messages":[{"content":"今天中午吃什么","role":"user"}]}' -H "Content-Type: application/json"
```

### CODE REVIEW

**Review by CH3CHO — CHANGES_REQUESTED** [review]

**Review by johnlanni — CHANGES_REQUESTED** [review]
vector 和 embeding 部分代码比较通用，建议从 ai-cache 目录下挪出来，放到 wasm-go 的 ai-utils 目录下

File: plugins/wasm-go/extensions/ai-cache/vector/chroma.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+			log.Infof("[Chroma] Query embedding response: %d, %s", statusCode, responseBody)
+			results, err := d.parseQueryResponse(responseBody, log)
+			if err != nil {
+				err = fmt.Errorf("[Chroma] Failed to parse query response: %v", err)
+			}
+			callback(results, ctx, log, err)
+		},
+		d.config.timeout,
+	)
+	return errors.New("[Chroma] QueryEmbedding Not implemented")
```
Comment (CH3CHO) [inline plugins/wasm-go/extensions/ai-cache/vector/chroma.go:92]: ？？

**Review by CH3CHO — APPROVED** [review]
LGTM

File: plugins/wasm-go/extensions/ai-cache/vector/weaviate.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	requestBody, err := json.Marshal(weaviateQueryRequest{
+		Query: graphql,
+	})
+
+	if err != nil {
+		log.Errorf("[Weaviate] Failed to marshal query embedding request body: %v", err)
+		return err
+	}
+
+	err = d.client.Post(
```
Comment (CH3CHO) [inline plugins/wasm-go/extensions/ai-cache/vector/weaviate.go:87]: SaaS 版本的 Weaviate 似乎需要一个 API Key 才能访问。

**Review by CH3CHO — CHANGES_REQUESTED** [review]

File: plugins/wasm-go/extensions/ai-cache/README.md [hunk 3]
```diff
@@ -101,29 +101,35 @@ LLM 结果缓存插件，默认配置方式可以直接用于 openai 协议的
...
+Pinecone 所对应的 `vector.type` 为 `pinecone`。它并无特有的配置字段。需要提前创建 Index，并填写 Index 访问域名至 `serviceHost`。
+Pinecone 中的 `Namespace` 参数通过插件的 `vector.collectionID` 进行配置。
 
 ## Qdrant
-Qdrant 所对应的 `vector.type` 为 `qdrant`。它并无特有的配置字段。
+Qdrant 所对应的 `vector.type` 为 `qdrant`。它并无特有的配置字段。需要提前创建 Collection。
 
 ## Weaviate
 Weaviate 所对应的 `vector.type` 为 `weaviate`。它并无特有的配置字段。
+需要提前创建 Collection。需要注意的是 Weaviate 会设置首字母自动大写，在填写配置 `collectionID` 的时候需要将首字母设置为大写。
```
Comment (CH3CHO) [inline plugins/wasm-go/extensions/ai-cache/README.md:131]: 这个 Weaviate 里好像是叫 class，不叫 collection。

Comment (CH3CHO) [reply plugins/wasm-go/extensions/ai-cache/README.md:131]: 其他几个 provider 的对应描述也调整一下吧。需要告知用户实际创建的是什么，需要把什么填入 collectionId 配置里。

**Review by CH3CHO — APPROVED** [review]
LGTM

**Review by johnlanni — APPROVED** [review]
LGTM

**Pull request merged**